src/oic/oauth2/provider.py

```diff
@@ -412,11 +413,15 @@ def auth_init(self, request, request_class=AuthorizationRequest):
             return error('unauthorized_client', 'unknown client')
         else:
             try:
-                rtypes = _cinfo['response_types']
-            except KeyError:
-                rtypes = ['code']  # default according to OIDC registration
-
-            if ' '.join(areq["response_type"]) not in rtypes:
+                _registered = [set(rt.split(' ')) for rt in
+                               _cinfo['response_types']]
+            except KeyError:  # default according to OIDC registration
+                _registered = [{'code'}]
+            except Exception as err:
+                raise
```

Comment:
Could we use a more descriptive error type here?

Reply:
Done